drake/systems/plants/constraint/MultipleTimeLinearPostureConstraint.m

```diff
@@ -1,33 +1,24 @@
 classdef MultipleTimeLinearPostureConstraint < RigidBodyConstraint
   % A linear constraint on the robot posture for multiple times
-  
+
   methods
-    function obj = MultipleTimeLinearPostureConstraint(robot,tspan)
-      if(nargin<2)
-        tspan = [-inf inf];
-      end
-      obj = obj@RigidBodyConstraint(RigidBodyConstraint.MultipleTimeLinearPostureConstraintCategory,robot,tspan);
+    function obj = MultipleTimeLinearPostureConstraint(robot)
+      obj = obj@RigidBodyConstraint(RigidBodyConstraint.MultipleTimeLinearPostureConstraintCategory,robot);
     end
-    
+
     function flag = isTimeValid(obj,t)
-      diff_t = diff(t);
-      if(any(diff_t)<=0)
-        error('Drake:MultipleTimeLinearPostureConstraint: t must be in the ascending order');
-      end
-      n_breaks = size(t,2);
-      if(n_breaks<=1)
-        error('Drake:MultipleTimeLinearPostureConstraint: t must have more than one entry');
-      end
-      flag = t>=obj.tspan(1) & t<=obj.tspan(end);
+      % TODO(sam.creasey) Understand the callers of this function well enough to get rid of it.
+      tspan = [-inf inf];
+      flag = t>=tspan(1)&t<=tspan(end);
     end
-    
+
     function [c,dc] = eval(obj,t,q)
       c = obj.feval(t,q);
       [iAfun,jAvar,A] = geval(obj,t);
       num_cnst = obj.getNumConstraint(t);
       dc = sparse(iAfun,jAvar,A,num_cnst,numel(q));
     end
-    
+
     function cnstr = generateConstraint(obj,t)
```

Comment:
`generateConstraint` should not take `t` as input argument. Instead it should take `num_knots` as argument. The only thing matters is the number of knot points, not the exact time of these knot points.